src/config_design.config

```diff
@@ -0,0 +1,126 @@
+   {get, 1},
+
+   %% Using alternate {op,[list]} format keeps 2-tuples so either format could be used
+   %% QUESTION: Do we need ONLY keygen or the OPTION of BOTH keygen and valgen ? 
+   %%    Would we ever need MORE per-op options? 
+
+   {put, [{count,1},  % Alternate count in more complex format, modify existing lookups
+          {key_generator, {int_to_str, {partitioned_sequential_int, 100000}}},
+          {value_generator, {function,http_couchdb_driver,make_json_doc,[]}}
+	 ]}
+   ]}.
+
+%% IMPLEMENTATION:
+%% basho_bench_worker:init 
+%%    turns per-op config into new Keygen/ValueGen entities
+%%    and stores into worker State as per-op-config list
+%%        per_op_config =[{put, [{keygen, KeyGen}, {valgen, ValGen}, ...]}, ... ]
+%% basho_bench_worker:worker_next_op2 
+%%    checks per_op_config, uses as run call arguments,
+%%       potentially stores onto State as current keygen/valgen
+
+%% ----------------------------------------------------------------------------
+%% WORKER_GROUP SUPPORT: https://github.com/cloudant/perf/issues/32
+%% ----------------------------------------------------------------------------
+%%
+%% "worker_types" would provide per-worker config list,
+%%     miniature copy of current config file for what is different from global
+
+{worker_types, [
+   %% Each worker_type can have its own list of config options
+   %%     to be used on top of global parameters known to basho_bench_config
+   {worker_name_a, [ {weight, 5}, 
+                     {driver, cb_driver}, 
+                     {operations, [
+                         {get,1},
+                         {put, [ {count,1}, 
+                                 {key_generator, keyGen1}
+                               ]}
+                         ]}, 
+                     {other, 1}
+                   ]},
+   {worker_name_b, []}
+   ]}.
+
+{workers, [
+   {worker_name_a, 1},
+   {worker_name_b, 3}
+   ]}.
+
+%% IMPLEMENTATION:
+%% The "concurrent" property picks how many workers but use workers weights for proportions needed. 
+%%
+%% basho_bench_worker_config:next_worker 
+%%    pops worker_name off weighted list based on "workers" proportions
+%%
+%% basho_bench_worker:init 
+%%    if next_worker is defined then 
+%%        looks up worker type to get per-worker-type config as LocalConfig
+%%        uses config_get(Key, LocalConfig) for initialization
+%%        stores LocalConfig into state for future use
+%%        benchmark applications can use config_get and then save values into own driverState
+%% 
+%% basho_bench_worker:config_get(Key,Default,LocalConfig) 
+%%    checks LocalConfig first then uses basho_bench_config:get for global defaults
+%%
+%% ISSUES
+%% 1. Some parms (source_dir, pre/post_hook, etc) and funs like log_dimensions may need more thought
+%%    to handle worker_types since these are currently handled outside basho_bench_worker.
+%%    May end up with list of pre- and post-hooks to run across multiple drivers, etc. 
+
```

Comment:
Once I got the basics working last night I realized a big issue we need to address is **basho_bench_stats** which also reads the operations list to initialize what folsom histogram metrics it will be keeping. 

Since we have to prepare for different operations for different workers AND for the possibility of the same driver being used for different worker_types that we want to keep separate, we need to decide how to deal with stats. 

My proposal would be that when worker_types are being used we automatically use the worker_type name as a stat prefix so instead of recording "get" and "put" we would record "worker_name_a_get" and "worker_name_a_put". In terms of implementation the stats could would need to do a preliminary sweep of the global operations list if provided AND the worker_types structure as actively used by workers (i.e. we don't want to initialize all POSSIBLE stats just those that could be generated by this run). 


Would that suffice or do we need an additional option to set the prefix more explicitly.